exercises/all-your-base/src/example.c

```diff
@@ -0,0 +1,37 @@
+#include "all_your_base.h"
+#include <math.h>
+
+size_t rebase(int16_t digits[DIGITS_ARRAY_SIZE], int16_t fromBase,
+              int16_t toBase, size_t numDigits)
+{
+   uint16_t denary = 0;
+   size_t newNumDigits = 0;
+
+   if ((fromBase <= 1) || (toBase <= 1) || (numDigits <= 0))
+      return 0;                 /* invalid bases or length */
+   if (digits[DIGITS_ARRAY_SIZE - numDigits] == 0)
+      return 0;                 /* leading zeros */
+
+   /* convert to denary */
+   for (size_t i = DIGITS_ARRAY_SIZE - 1, j = 0;
+        i >= DIGITS_ARRAY_SIZE - numDigits; --i, ++j) {
+      if (digits[i] < 0)
+         return 0;              /* negative digits */
+      if (digits[i] >= fromBase)
+         return 0;              /* invaid positive digit */
+      denary += digits[i] * pow(fromBase, j);
+   }
+
+   /* calculate and store new digits */
+   for (size_t i = DIGITS_ARRAY_SIZE - 1; denary > 0; --i) {
+      digits[i] = denary % toBase;
+      denary /= toBase;
+      ++newNumDigits;
+   }
+
+   /* ensure rest of array is zero */
+   for (int i = DIGITS_ARRAY_SIZE - (newNumDigits + 1); i >= 0; --i)
+      digits[i] = 0;
```

Comment:
BTW, would `memset()` actually have worked here? The data type of the array being `int` at the time.
It's changed to `uint8_t` now so it works now, but `memst()` is designed for strings and writes characters, so I'm guessing as part of it's optimisation it would at the least only end up clearing half the length of the `int` array?